test-e2e/sync.js

```diff
@@ -259,3 +260,40 @@ test('no sync capabilities === no namespaces sync apart from auth', async (t) =>
 
   await Promise.all(projects.map((p) => p.close()))
 })
+
+test('Sync state emitted when starting and stopping sync', async function (t) {
```

Comment:
potential suggestion: add some assertions to test idempotency of `start()` and `stop()` i.e. events are not emitted for subsequent calls that do not meaningfully change sync state

maybe that's diving too much into internals though...